src/bitcoin/Bitcoin.mo

```diff
@@ -118,10 +120,23 @@ module {
           transaction.txInputs.size(), func (i) {
             let sighash : [Nat8] = transaction.createSignatureHash(
               scriptPubKey, Nat32.fromIntWrap(i), Types.SIGHASH_ALL);
+              let signature : Blob = ecdsaProxy.sign(Blob.fromArray(sighash), []);
+              let encodedSignature : [Nat8] = Blob.toArray(
+                Der.encodeSignature(signature));
+              // Append the sighash type.
+              let encodedSignatureWithSighashType = Array.tabulate<Nat8>(
```

Comment:
I initially thought about:
```
let encodedSignatureWithSighashType = Array.append<Nat8>(encodedSignature, [Nat8.fromNat(Nat32.toNat(Types.SIGHASH_ALL))]);
```
Is there any advantage with the code you are using, like complexity ?